services/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -41,7 +43,11 @@ object MetadataServiceActor {
   case class WorkflowOutputs(workflowId: WorkflowId) extends MetadataServiceAction
   case class GetLogs(workflowId: WorkflowId) extends MetadataServiceAction
   case object RefreshSummary extends MetadataServiceAction
-  final case class HandleNotFound(workflowId: WorkflowId, sndr: ActorRef) extends MetadataServiceAction
+  final case class ValidateWorkflowIdAndExecute(uuid: UUID,
+                                                requestContext: RequestContext,
+                                                onRecognized: WorkflowId => Route,
+                                                onUnrecognized: UUID => Route,
+                                                onFailure: (UUID, Throwable) => Route) extends MetadataServiceAction
```

Comment:
Two things related to loving the distinction of `unrecognized` vs `malformed` back in the web service:

1) Have you considered using a single trait instead of three function parameters? If this is your preference, then that's fine.

2) What about changing `uuid: UUID` to `possibleWorkflowId: String`, and adding to the class a `def workflowId`. This would require a fourth `onMalformed: (String, Throwable) => Route` function, that made me think of "1)" above.

There's something about UUID imports leaking around, and what seems to be even the tiniest biz logic up in the web services that feels funny to me. This `ValidateWorkflowIdAndExecute` is already doing other validation, so I'm wondering if it could also start validation with the string.